File: SkillSync.Client/src/app/shared/entities/responses/freelancer-dto.ts
import { User } from "../models/user";
import { FreelancerReviewsDto } from "./freelancer-reviews-dto";
import { FreelancerSkillsDto } from "./freelancer-skills-dto";
import { GetProjectPreviewResponse } from "./get-projects-preview-response";

export interface FreelancerDto extends User {
    id: string,
    description: string,
    countryCode: string,
    skills: FreelancerSkillsDto[],
    projects: GetProjectPreviewResponse[],
    reviews: FreelancerReviewsDto
}

File: SkillSync.WebServer/SkillSync.Domain/DTOs/Models/FreelancerDto.cs
using SkillSync.Domain.DTOs.Project;
using SkillSync.Domain.DTOs.Review;

namespace SkillSync.Domain.DTOs.Models
{
    public class FreelancerDto
    {
        public string Id { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public string Description { get; set; }

        public string CountryCode { get; set; }

        public string ProfilePicture { get; set; }

        public List<FreelancerSkillsDto>? Skills { get; set; }

        public List<GetProjectPreviewDto> Projects { get; set; }

        public FreelancerReviewsDto? Reviews { get; set; }
    }

    public class FreelancerSkillsDto
    {
        public string SkillName { get; set; }
        public Guid SkillId { get; set; }
    }
}